contracts/token/ERC1155/extensions/ERC1155Supply.sol

```diff
@@ -41,21 +49,30 @@ abstract contract ERC1155Supply is ERC1155 {
         bytes memory data
     ) internal virtual override {
         if (from == address(0)) {
+            uint256 totalMintAmount = 0;
             for (uint256 i = 0; i < ids.length; ++i) {
-                _totalSupply[ids[i]] += amounts[i];
+                uint256 amount = amounts[i];
+                _totalSupply[ids[i]] += amount;
+                totalMintAmount += amount;
             }
+            _totalSupplyAll += totalMintAmount;
         }
 
         if (to == address(0)) {
+            uint256 totalBurnAmount = 0;
             for (uint256 i = 0; i < ids.length; ++i) {
                 uint256 id = ids[i];
                 uint256 amount = amounts[i];
                 uint256 supply = _totalSupply[id];
                 require(supply >= amount, "ERC1155: burn amount exceeds totalSupply");
                 unchecked {
                     _totalSupply[id] = supply - amount;
+                    totalBurnAmount += amount;
```

Comment:
It isn't obvious to me why this variable won't overflow. We need to add the explanation.

Reply:
It comes from two inequalities:
- `amounts[i] <= balanceOf(from, i)` (from the require on line 67)
- `sum(balanceOf(from, i) <= sum of all balances = totalSupply` (contract invariant)

so `totalBurnAmount = sum(amounts[i]) <= sum(balanceOf(from, i) <= totalSupply <= 2*256-1`

Reply:
Technically we could even get rid of the check on line 67 if we were to check `balance >= amount` before. (the check if done after, because this is the "pre" hook)

Reply:
@Amxx I think we didn't do that because we were concerned what would happen if ERC1155Supply was added in an upgrade...